### PR TITLE
Salt'ify mark-control-plane phase

### PR DESCRIPTION
**Pull request by gdemonet**
Note that we also added an annotation for specifying the CRI socket used, as can be seen in the [kubeadm dry-run output](https://github.com/NicolasT/kubeadm-dry-run/blob/master/kubeadm-dry-run.txt#L518-L524).

Fixes GH-582.

### CODE REVIEW

**Comment by NicolasT** [comment]
@bert-e wait

**Comment by NicolasT** [comment]
Let's wait for CI to merge #689, then rebase this one on `development/2.0` and retarget this PR to `development/2.0` (no need to close this PR, can edit it in-place).

File: salt/_modules/kubernetes.py
Comment (NicolasT) [inline salt/_modules/kubernetes.py]: Is this done in the other procedures? There should be no need, either `return api_response` is reached, or the `return None` or `raise` in the `except` clause, or some other exception bubbles up. This is unreachable.

Also applies to other occurrences.

File: salt/_modules/kubernetes.py
Comment (NicolasT) [inline salt/_modules/kubernetes.py]: This means the node doesn't exist, rather than 'no taints'. If this is how other procedures in the module work, then fine. We should rework it anyway :stuck_out_tongue:

File: salt/_states/kubernetes.py
Comment (NicolasT) [inline salt/_states/kubernetes.py:1092]: In this code-path, `ret['comment']` is not set.

File: salt/_states/kubernetes.py
Comment (NicolasT) [inline salt/_states/kubernetes.py]: I'd prefer no docstring (which can be caught by static analysis) than a docstring which is de-facto empty, no?

File: salt/_states/kubernetes.py
Comment (NicolasT) [inline salt/_states/kubernetes.py]: Prefer not to call things with side-effects in a comprehension.

File: salt/_states/kubernetes.py
Comment (NicolasT) [inline salt/_states/kubernetes.py:1173]: `ret['comment']` not set

File: salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls
Comment (NicolasT) [inline salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls:8]: @TeddyAndrieux prefers `name` to go first, then `kubeconfig` and `context`, then `node` and `value`. I kinda agree (had to be convinced though ;))

File: salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls
Comment (NicolasT) [inline salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls]: Same

File: salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls
Comment (NicolasT) [inline salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls]: `TODO` i.s.o. `NOTE`.

**Review by NicolasT — CHANGES_REQUESTED** [review]
Definitely requires rebase after #685 lands.

File: salt/_modules/kubernetes.py
Comment (gdemonet) [reply salt/_modules/kubernetes.py]: Indeed, not sure exactly why I added them, will remove when rebasing.

File: salt/_modules/kubernetes.py
Comment (gdemonet) [reply salt/_modules/kubernetes.py]: Actually I should have returned an empty list instead (copy-pasted the None as for the other methods I guess), but yes this module needs a rework :)

File: salt/_states/kubernetes.py
Comment (gdemonet) [reply salt/_states/kubernetes.py]: Agreed, I was hoping to find the courage to fill it before pushing...

File: salt/_states/kubernetes.py
Comment (gdemonet) [reply salt/_states/kubernetes.py:1092]: Done.

File: salt/_states/kubernetes.py
Comment (gdemonet) [reply salt/_states/kubernetes.py]: Split the comprehension in two.

File: salt/_states/kubernetes.py
Comment (gdemonet) [reply salt/_states/kubernetes.py:1173]: Done.

File: salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls
Comment (gdemonet) [reply salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls:8]: Done.

File: salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls
Comment (gdemonet) [reply salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls]: Done.

File: salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls
Comment (gdemonet) [reply salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls]: Done.

**Comment by gdemonet** [comment]
Rebased and integrated review comments from @NicolasT

**Comment by gdemonet** [comment]
Build failed since `mark-control-plane` runs before `addons`, and as such `python2-kubernetes` is not yet installed when we execute this step.

A quick fix could be to include `..addons.dependencies` in `metalk8s.kubeadm.init.mark-control-plane.init`, but this doesn't look "clean".
I'm wondering if doing something like a `metalk8s.python-kubernetes` formula could be useful / acceptable.

File: salt/_modules/kubernetes.py
Comment (cmonfort) [inline salt/_modules/kubernetes.py:2221]: Is the annotation removed when its value is set to `None`?
If not, it's not really a removal, and if I get it afterwards, I will not have a failure because it does not exist, but retrieve a `None` value...)

File: salt/_states/kubernetes.py
Comment (cmonfort) [inline salt/_states/kubernetes.py:1116]: Get the full annotations block? not only the value of one we are changing?

File: salt/_modules/kubernetes.py
Comment (cmonfort) [inline salt/_modules/kubernetes.py:2096]: For consistency with other annotations commands, `name` parameter should be `node_name`.

Comment (gdemonet) [reply salt/_modules/kubernetes.py:2096]: I copy-pasted the signature of [`node_labels`](https://github.com/scality/metalk8s/blob/020703de54117108286831e813d3076ecdf39d3a/salt/_modules/kubernetes.py#L321), I'll keep it as is until we refactor this custom module (which is essentially a long list of copy-pasted functions)

File: salt/_modules/kubernetes.py
Comment (gdemonet) [reply salt/_modules/kubernetes.py:2221]: Yup, setting an annotation or label to `None` in a PATCH call will effectively remove it - setting a label/annotation with _no value_ would be done by using an empty string instead.

File: salt/_states/kubernetes.py
Comment (gdemonet) [reply salt/_states/kubernetes.py:1116]: Same comment as for the signature of `node_annotations`, I copy-pasted what was done for labels, and they indeed show the whole labels block instead of simply the one changed... To discuss in future improvements.

**Review by cmonfort — DISMISSED** [review]

**Review by NicolasT — DISMISSED** [review]
LGTM (didn't run)

**Comment by gdemonet** [comment]
@bert-e approve

**Review by charlesprost — APPROVED** [review]

File: salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls
Comment (ramanduh) [inline salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls:29]: I think, it's already configurable in default.yaml: `kubelet.service.options.container-runtime-endpoint`

**Pull request merged**
